MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -42,6 +41,53 @@ final class MapViewModel: ObservableObject {
         CLLocationManager().requestWhenInUseAuthorization()
         CLLocationManager().desiredAccuracy = kCLLocationAccuracyBest
     }
+
+    func onNearbyFarmerButtonTapped() {
+        do {
+            if let nearbyFarmer = try searchNearbyFarmer() {
+                let nearbyFarmerCoordinate = CLLocationCoordinate2D(latitude: nearbyFarmer.location.latitude,
+                                                                    longitude: nearbyFarmer.location.longitude)
+                let nearbyFarmerRegion = MKCoordinateRegion(center: nearbyFarmerCoordinate,
+                                                            span: self.nearbyFarmerSpan)
+                mapCameraPosition = .region(nearbyFarmerRegion)
+            }
+        } catch {
+            print("Aucun maraîcher trouvé à proximité")
+        }
+    }
+
+    private func searchNearbyFarmer() throws -> Farmer? {
+        var maxSearchDistance = 10_000.0
+        var nearbyFarmer: Farmer?
+        guard let userLocation = CLLocationManager().location else { return nil }
+        for farmer in allFarmers {
+            let distance = userLocation.distance(from: .init(latitude: farmer.location.latitude, longitude: farmer.location.longitude))
+            if distance < maxSearchDistance {
```

Comment:
I will show you how you can write that logic more "Swifty" ;)

Reply:
I look forward to